quint/src/parsing/ToIrListener.ts

```diff
@@ -347,14 +347,33 @@ export class ToIrListener implements QuintListener {
     const id = this.idGen.nextId()
     this.sourceMap.set(id, this.loc(ctx))
 
-    const def: QuintDef = {
+    const def: QuintTypeDef = {
       id,
       kind: 'typedef',
       name,
     }
 
-    if (typeToAlias.isJust()) {
-      def.type = typeToAlias.value
+    this.definitionStack.push(def)
+  }
+
+  // type Alias = set(int)
+  exitTypeAliasDef(ctx: p.TypeAliasDefContext) {
+    const name = ctx.IDENTIFIER()!.text
+    const type = this.popType().value
+
+    if (name[0].match('[a-z]')) {
+      const msg = 'QNT007: type names must start with an uppercase letter'
+      this.pushError(ctx, msg)
+    }
+
+    const id = this.idGen.nextId()
+    this.sourceMap.set(id, this.loc(ctx))
+
+    const def: QuintTypeDef = {
+      kind: 'typedef',
+      id,
```

Comment:
Probably nitpicking here, I'm never certain how TS handles object fields given by order rather than by name:

The prevalent style I've seen in Quint is `id` first, `kind` second: 

```suggestion
      id,
      kind: 'typedef',
```